Character.cpp

```diff
@@ -86,32 +89,36 @@ int Character::takeDamage(int damage)
     return hitPoints;
 }
 
-
-#include <assert>
+#include <cassert>
 void Character::attackInternal(Character& other)
 {
-    if( other.hitPoints <= 0 )
+    if (other.hitPoints <= 0)
     {
         /*
-        When you defeat another Character: 
+        When you defeat another Character:
             a) your stats are restored to their initial value
             b) your stats are boosted 10%
-            c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
+            c) the initial value of your stats is updated to reflect this
+        boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        hitPoints = static_cast<int>(*initialHitPoints * 1.1);
+        armor = static_cast<int>(*initialArmorLevel * 1.1);
+        attackDamage = static_cast<int>(*initialAttackDamage * 1.1);
+
+        initialHitPoints = std::make_unique<int>(hitPoints);
+        initialArmorLevel = std::make_unique<int>(armor);
+        initialAttackDamage = std::make_unique<int>(attackDamage);
```

Comment:
You don't need to make new instances of these.  Dereference the pointer to update
- [ ] FIXME